### PR TITLE
Add plugin review guidelines

### PR DESCRIPTION
**Pull request by ehmicky**
This adds guidelines on how to review plugins. This is primarily intended for Netlify reviewers, while `guidelines.md` is mostly intended for plugin authors.

@erezrokah Please share any feedback!

### CODE REVIEW

File: docs/plugin_review.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Utilities
+
+- [ ] The [`cache` utility](https://github.com/netlify/build/blob/master/packages/cache-utils/README.md) should be used to cache files.
+- [ ] The [`git` utility](https://github.com/netlify/build/blob/master/packages/git-utils/README.md) should be used for git-related logic.
+
+## Control flow
+
+- [ ] Asynchronous code should use `async`/`await`.
+- [ ] Plugin methods [should not end until all their asynchronous code has completed](https://docs.netlify.com/configure-builds/build-plugins/create-plugins/#asynchronous-code). Also, asychronous should propagate to the upper scopes. When mixing callbacks, events and promises, it is common to miss this. Using promise-friendly libraries and `utils.promisify()` are common solutions.
```
Comment (erezrokah) [inline docs/plugin_review.md:97]:
```suggestion
- [ ] Plugin methods [should not end until all their asynchronous code has completed](https://docs.netlify.com/configure-builds/build-plugins/create-plugins/#asynchronous-code). Also, asychronous should propagate to the upper scopes. When mixing callbacks, events and promises, it is common to miss this. Using promise-friendly libraries and `util.promisify()` are common solutions.
```

**Review by erezrokah — APPROVED** [review]
Hi @ehmicky this is great :)
Do you think we should emphasize that UI plugins must allow zero config?
It's written un the `input` section, but maybe make it clear that this is a blocker.

**Comment by ehmicky** [comment]
Don't we actually allow UI plugins with required environment variables like API tokens, providing there's no way around it?

**Comment by erezrokah** [comment]
> Don't we actually allow UI plugins with required environment variables like API tokens, providing there's no way around it?

Yes we do. Do you think we should make it more visible that this should be a last resort (we can provide an API token env variable as an example of something that is unavoidable).

**Comment by ehmicky** [comment]
Done in 30c24eb94e73e579f2bc2e49c22d85195f66aa9b, please let me know what you think.

**Review by erezrokah — APPROVED** [review]

**Pull request merged**